### PR TITLE
Added a similarity threshold to avoid outputting identical communities

### PR DESCRIPTION
**Pull request by pabloi09**

### CODE REVIEW

File: scripts/blant-clusters.sh [hunk 1]
```diff
@@ -170,7 +172,6 @@ for k in "${Ks[@]}"; do
 		for(start=1; start<=FNR; start++) { # look for a cluster starting on line "start". 
 		    if(QueueLength("Q")>0){QueueDelloc("Q");QueueAlloc("Q");} #Ensure the queue is empty
 		    origin=node[start];
-		    if (origin in visited) continue;
```
Comment (waynebhayes) [inline scripts/blant-clusters.sh:173]: Hmm. Why is this line deleted?

File: scripts/blant-clusters.sh [hunk 2]
```diff
@@ -212,31 +213,53 @@ for k in "${Ks[@]}"; do
...
+		}
+		if(numCliques==0 || same < length(S)*'$t') {
+			    maxEdges=choose(length(S),2);
+			    ++numCliques; 
+				printf "%d nodes, %d of %d edges from k %d (%g%%):",
+			length(S), edgeHits, maxEdges, k, 100*edgeHits/maxEdges
+			    for(u in S) {cluster[numCliques][u]=1; printf " %s", u}
+			    print ""
+		}
+	    }' - | sort -k 1nr -k 11n # sort by number of nodes and then by the first node in the list
```
Comment (waynebhayes) [inline scripts/blant-clusters.sh:262]: I think the second sort should be changed to be similar to the one above, to sort by column 3 rather than 11??

File: scripts/blant-clusters.sh [hunk 3]
```diff
@@ -212,31 +213,53 @@ for k in "${Ks[@]}"; do
...
-			    if(same == length(S)) break;
+			    if(same >= length(S)*'$t') break;
 		}
-		if(numCliques==0 || same < length(S)) {
+		if(numCliques==0 || same < length(S)*'$t') {
 			    maxEdges=choose(length(S),2);
-			    ++numCliques; printf "%d nodes, %d of %d edges from k '$k' (%g%%):",
-			length(S), edgeHits, maxEdges, 100*edgeHits/maxEdges
+			    ++numCliques; 
+				printf "%d %d '$k'",length(S),edgeHits
```
Comment (waynebhayes) [inline scripts/blant-clusters.sh:229]: I'm OK with shortening the output a bit, but the edge density I think needs to be printed.. plus please leave the colon before the list of nodes, I need that to be able to easily isolate the preamble from the list of nodes.

**Review by waynebhayes — COMMENTED** [review]
Please see comments inline.

File: scripts/blant-clusters.sh (hunk 1 above)
Comment (pabloi09) [reply scripts/blant-clusters.sh:173]: This was (unsuccessfully) trying to avoid very similar communities that come next to each other. The tradeoff was we could miss a bigger result. Now very similar communities are handled later in the code

File: scripts/blant-clusters.sh (hunk 3 above)
Comment (pabloi09) [reply scripts/blant-clusters.sh:229]: Second output is different to first output. I shorten the first output because we don't need it formatted yet.  The first output is only used internally. The final output stays the same as in previous version.

Comment (waynebhayes) [reply scripts/blant-clusters.sh:229]: AHA! OK. Got it. Thanks.

**Review by waynebhayes — APPROVED** [review]
